test/e2e/network_test.go

```diff
@@ -301,12 +302,25 @@ func (tc *testContext) testEastWestNetworking(t *testing.T) {
 
 	linuxServerDeployment, err := tc.deployLinuxWebServer()
 	require.NoError(t, err)
-	defer tc.deleteDeployment(linuxServerDeployment.GetName())
 	linuxServerClusterIP, err := tc.createService(linuxServerDeployment.GetName(), 8080, v1.ServiceTypeClusterIP,
 		*linuxServerDeployment.Spec.Selector)
 	require.NoError(t, err)
-	defer tc.deleteService(linuxServerClusterIP.GetName())
-	linuxServerIP, err := tc.getPodIP(*linuxServerDeployment.Spec.Selector)
+	// best effort to collect Linux web server logs before deleting the deployment and service
+	defer func(tc *testContext, deployment *appsv1.Deployment) {
+		err := tc.collectDeploymentLogs(deployment)
+		if err != nil {
+			log.Printf("error collecting deployment logs: %v", err)
+		}
+		err = tc.deleteDeployment(linuxServerDeployment.GetName())
+		if err != nil {
+			log.Printf("error deleting deployment: %v", err)
+		}
+		err = tc.deleteService(linuxServerClusterIP.GetName())
```

Comment:
This should not be in the block, only the collectLogs +deleteDeployment.
These should be defered directly after the object they are cleaning up is created.

order to not leak anything:
```
create service
fail and exit if service not created
defer service deletion
create deployment
fail and exit if deployment not created
defer log collection and deletion for deployment

```

Reply:
addressed https://github.com/openshift/windows-machine-config-operator/compare/93ad6f8c873b8922a8e0ed49493d485d51a77626..4e43656d6d5e8209f983c121ceef6b1967a5d86d